internal/services/reservations_service.go

```diff
@@ -23,6 +23,7 @@ var (
 	ErrBothTypeAndTemplateMissing = errors.New("instance type or launch template not set")
 	ErrUnsupportedRegion          = errors.New("unknown region/location/zone")
 	ErrInvalidNamePattern         = errors.New("name pattern is not RFC-1035 compatible")
+	ErrPubkeyNotFound             = errors.New("no pubkey found, it may have been already deleted")
```

Comment:
For plain Go errors I would keep it simple:

```suggestion
	ErrPubkeyNotFound             = errors.New("no pubkey found")
```

Only in "user message" (above) it makes sense I think.